ext/span.c

```diff
@@ -796,3 +823,75 @@ zend_string *ddtrace_trace_id_as_hex_string(ddtrace_trace_id id) {
     snprintf(ZSTR_VAL(str), 33, "%016" PRIx64 "%016" PRIx64, id.high, id.low);
     return str;
 }
+
+
+ddtrace_root_span_data *ddtrace_open_inferred_span(zend_array *headers) {
+    ddtrace_read_header *read_header = headers ? ddtrace_read_array_header : ddtrace_read_zai_header;
+    ddtrace_inferred_proxy_result result = ddtrace_read_inferred_proxy_headers(read_header, headers);
+
+    if (!result.system || !result.start_time_ms) {
+        if (result.system) zend_string_release(result.system);
+        if (result.start_time_ms) zend_string_release(result.start_time_ms);
+        return NULL;
+    }
+
+    const ddtrace_proxy_info *proxy_info = ddtrace_get_proxy_info(result.system);
+    if (!proxy_info) {
+        zend_string_release(result.system);
+        zend_string_release(result.start_time_ms);
+        return NULL;
```

Comment:
You're leaking the other strings in result here, there are more than two.

Reply:
Maybe you should just `goto cleanup;` with the cleanup label at the end of the function here.